lib/util.js

```diff
@@ -78,4 +63,63 @@ util.findToken = function (req) {
   return token
 }
 
+function sizeof (obj) {
+  if (obj === undefined || obj === null) return 0
+  if (isNumber(obj)) return sizeChart.NUMBER
+  if (isBoolean(obj)) return sizeChart.BOOLEAN
+  if (isString(obj)) return sizeChart.STRING * obj.length
+
+  if (!isPlainObject(obj) &&
+    !Array.isArray(obj)) {
+    return -1
+  }
+
+  var size = 0
+  for (var key in obj) {
+    if (obj.hasOwnProperty(key)) {
+      size += sizeof(key)
+      try {
```

Comment:
plz remove this try/catch. I don't think this is a good way of solving the curcular ref problem. track separately in aha